### ISSUE TITLE
Abnormality of lingual tonsil (transitively) a subclass of Abnormality of face

### ISSUE THREAD

**Issue by cmungall** [obophenotype/upheno#957]
hacking session with @julesjacobsen we noticed an oddity:

<img width="188" alt="image" src="https://user-images.githubusercontent.com/50745/168346156-e744e9fd-3086-4f73-808d-1301f5523528.png">

Is there a TPV on the left side here?

here it is as a tree:

    * [i] HP:0000118 ! Phenotypic abnormality
        * [i] HP:0000152 ! Abnormality of head or neck
            * [i] HP:0000234 ! Abnormality of the head
                * [i] HP:0000271 ! Abnormality of the face
                    * [i] HP:0000153 ! Abnormality of the mouth
                        * [i] HP:0031816 ! Abnormal oral morphology
                            * [i] HP:0000163 ! Abnormal oral cavity morphology
                                * [i] HP:0000157 ! Abnormality of the tongue
                                    * [i] HP:0030809 ! Abnormal tongue morphology
                                        * [i] HP:3000076 ! Abnormality of lingual tonsil

Each individual link seems justifiable but transitively the link from tonsil to face doesn't seem right. I think the problem here is terms like oral cavity and mouth can be contextual

(btw this is the current asserted structure, not inferred)

**Comment by rays22** [comment]
I have made a ticket in Uberon to reconsider the partonomy of the tongue and the oral cavity, which could potentially fix the inference issues with [HP:3000076 Abnormality of lingual tonsil](https://ols.monarchinitiative.org/ontologies/upheno2/terms?iri=http%3A%2F%2Fpurl.obolibrary.org%2Fobo%2FHP_3000076).

Alternatively, modifying the EQ of [HP:0000163 Abnormal oral cavity morphology](https://www.ebi.ac.uk/ols4/ontologies/hp/classes/http%253A%252F%252Fpurl.obolibrary.org%252Fobo%252FHP_0000163?lang=en) could fix the issue in HPO and uPheno without restructuring Uberon. Replace the `characteristic of part of` property chain with `characteristic of` for more intuitive phenotype inferences:
* Suggested EQ:

```
'has part' some (morphology
    and ('characteristic of' some 'oral cavity')
    and ('has modifier'  some abnormal))
```

**Comment by rays22** [comment]
> Probably mouth is not part of face and the face stops at the vermilion border! Need to figure out a better placement.

1. I can confirm that [UBERON:0000165 mouth](http://purl.obolibrary.org/obo/UBERON_0000165) is not part of [UBERON:0001456 face](http://purl.obolibrary.org/obo/UBERON_0001456). Revising the HPO subclass assertion that an `Abnormality of the mouth` is an `Abnormality of the face` would be part of the resolution of this issue.
2. I can confirm that [UBERON:0001723 tongue](http://purl.obolibrary.org/obo/UBERON_0001723) is not part of [UBERON:0000167 oral cavity](http://purl.obolibrary.org/obo/UBERON_0000167). Revising the HPO subclass assertion that an 'Abnormality of the tongue` is an `Abnormal oral cavity morphology` would be the second part of the resolution of this issue.

In summary, this issue stems from subclass assertions in HPO, and it is not an issue of the Uberon class hierarchy. A computed phenotype class hierarchy that is based on Uberon would not show the [inferences above.](https://github.com/obophenotype/upheno/issues/957#issue-2407313268)

**Comment by balhoff** [comment]
> 1. I can confirm that [UBERON:0000165 mouth](http://purl.obolibrary.org/obo/UBERON_0000165) is not part of [UBERON:0001456 face](http://purl.obolibrary.org/obo/UBERON_0001456). Revising the HPO subclass assertion that an `Abnormality of the mouth` is an `Abnormality of the face` would be part of the resolution of this issue.

However Uberon does say that every [face](http://purl.obolibrary.org/obo/UBERON_0001456) has_part some [mouth](http://purl.obolibrary.org/obo/UBERON_0000165). So any instance model in which there is a face (like say a human) would imply that the mouth is part of the face (inverse properties). So the HPO subclass relationship is consistent with this.

**Comment by rays22** [comment]
> > 1. I can confirm that [UBERON:0000165 mouth](http://purl.obolibrary.org/obo/UBERON_0000165) is not part of [UBERON:0001456 face](http://purl.obolibrary.org/obo/UBERON_0001456). Revising the HPO subclass assertion that an `Abnormality of the mouth` is an `Abnormality of the face` would be part of the resolution of this issue.
> 
> However Uberon does say that every [face](http://purl.obolibrary.org/obo/UBERON_0001456) has_part some [mouth](http://purl.obolibrary.org/obo/UBERON_0000165). So any instance model in which there is a face (like say a human) would imply that the mouth is part of the face (inverse properties). So the HPO subclass relationship is consistent with this.

Good point @balhoff .
There is a new [ticket in Uberon to make the lip ( (located at the opening of the mouth) part of the face in Uberon.](https://github.com/obophenotype/uberon/issues/3311)  This should allow us to drop the [face](http://purl.obolibrary.org/obo/UBERON_0001456) 'has part' some mouth' assertion in Uberon, because the lips would represent the mouth-specific parts that are visible as also parts of the face in contrast to structures that are deeper in the mouth.

**Comment by rays22** [comment]
This is to [summarise some of the 'part of' and 'has part' relationships in Uberon resulting from Uberon issue #3311.](https://github.com/obophenotype/uberon/issues/3311)
* The [lingual tonsil](http://purl.obolibrary.org/obo/UBERON_0007250) is no longer part of the [face](http://purl.obolibrary.org/obo/UBERON_0001456) (transitively) in Uberon, and the
* [face](http://purl.obolibrary.org/obo/UBERON_0001456) does not have the [mouth](http://purl.obolibrary.org/obo/UBERON_0000165) as its part.
* Now the [lip](http://purl.obolibrary.org/obo/UBERON_0001833) is asserted to be part of the [face](http://purl.obolibrary.org/obo/UBERON_0001456) in Uberon.
* The [lip](http://purl.obolibrary.org/obo/UBERON_0001833) remains also part of the [mouth](http://purl.obolibrary.org/obo/UBERON_0000165), and the
*  [lingual tonsil](http://purl.obolibrary.org/obo/UBERON_0007250) also remains part of the [mouth](http://purl.obolibrary.org/obo/UBERON_0000165) in Uberon.

In summary, *automatic reasoning should no longer infer that the [lingual tonsil](http://purl.obolibrary.org/obo/UBERON_0007250) is part of the [face](http://purl.obolibrary.org/obo/UBERON_0001456).* However, there is still at least one subclass assertion in HPO that needs to be revised to avoid inferring that an 'Abnormality of the lingual tonsil' is also an 'Abnormality of the face' phenotype.

**Comment by matentzn** [comment]
Thank you @rays22 for working on this and seeing this through! Much appreciated!
